modules/libcom/src/osi/os/vxWorks/osdEnv.c

```diff
@@ -52,8 +52,9 @@ LIBCOM_API void epicsStdCall epicsEnvSet (const char *name, const char *value)
         errPrintf(-1L, __FILE__, __LINE__,
             "Failed to set environment parameter \"%s\" to \"%s\": %s\n",
             name, value, strerror (errno));
-        free (cp);
     }
+    /* from at least vxWorks 5.5 putenv() is making a copy, so we can free */
+    free (cp);
```

Comment:
It would be nice to make `cp` a stack variable here, thus completely avoiding the need to call `malloc()` and `free()`. Since we're using gcc the following should be legal (but we still have to abide by the 'allocations before statements' rule so it must appear at the start of a block):

```C
    iocshEnvClear(name);
    {
        char cp[strlen (name) + strlen (value) + 2];

        strcpy (cp, name);
        strcat (cp, "=");
        strcat (cp, value);
        if (putenv(cp) < 0) {
```
The "Leaks memory" comment higher up could also be deleted since we won't any more.